cEP-0025.md

```diff
@@ -0,0 +1,169 @@
+# Integration of pyflakes-enhanced-AST into coala
+
+| Metadata |                                                 |
+| -------- | ----------------------------------------------- |
+| cEP      | 22                                              |
+| Version  | 1.0                                             |
+| Title    | Integration of pyflakes-enhanced-AST into coala |
+| Authors  | Ankit Joshi <mailto:ajankit2304@gmail.com>      |
+| Status   | Proposed                                        |
+| Type     | Feature                                         |
+
+## Abstract
+
+This document describes how to create a metabear to integrate
+pyflakes-enhanced-AST into coala and demonstrates its use by
+creating a simple dependent bear.
+
+## Introduction
+
+Flake8 is the most commonly used linter framework when it comes to static
+code analysis for python. It basically wraps around the tools pyflakes,
+pycodestyle and mccabe scripts because each of those projects
+tend to add new checks in minor releases. That way if you install something
+from the 2.5 series today, you will not find new checks in the same series
+in a month from now when you install it again.
+
+It even provides with a plugin mechanism. Additional linters are
+supported using the flake8 plugin system, and may be Python AST
+based or built using pycodestyle's internal logical-line-based
+checker API (checkout [Hacking](https://github.com/openstack-dev/hacking)).
+
+There has not been an option to use the pyflakes internal AST based checker
+API to create custom linters. Thus, the full potential of enhanced AST isn't
+utilized , a whole lot of rework is required to do the basic traversing
+and collection of important nodes. Pyflakes provides with a basic API
+that does the traversing. So, if a developer uses enhanced AST he just
+needs to work on the implementation of the new logic that his/her plugin
+provides and not about the fidelity of the basic node handlers.
+
+The reason for choosing pyflakes over its equivalents like
+[Asteroid](https://github.com/PyCQA/astroid) lies in its simplicity
+and provision of lots of re-usable analysis. Asteroid is more detailed and
+powerful, thus making it complicated. Plus, there is a much larger ecosystem
+of flake8 plugins.
+
+## Proposed Change
+
+Here is a brief overview of the architecture:
+
+1. There will be a separate repository named as coala-pyflakes which will
+   be installable via `pip install coala-pyflakes`.
```

Comment:
It seems like we have three packages in one repo.

- pyflakes-generic-plugins is easy -- it contains the generic plugins, and other generic stuff
- pyflakes-plugin-bears is pyflakes enhanced AST coala Bears.

If `pyflakes-plugin-bears` contains metabear `PyFlakesASTBear` ...

What is in `coala-pyflakes` ?